crates/acvm/src/lib.rs

```diff
@@ -30,6 +31,7 @@ pub enum GateResolution {
     UnknownError(String),      //Generic error
     UnsupportedOpcode(OPCODE), //Unsupported Opcode
     UnsatisfiedConstrain,      //Gate is not satisfied
+    Solved(usize),             //Circuit is solved, after a number of passes
```

Comment:
It is useful at least for debugging purposes and it is harmless to report it.

Reply:
I don't see where we're reporting it but if its only for debug information I'd rather we leave it out since it isn't strictly necessary.

Reply:
I prefer to keep it, because of its negligible impact.

Reply:
I still think we should remove it since it complicates understanding of the code a bit, as readers will now encounter this extra variant and try to debug/trace it only to later discover it only holds debug information and is otherwise covered by the Resolved case. In the event the number of passes needs to be debugged it should be simple enough to add a quick println even without this variant.

Reply:
Done